src/librustc/infer/error_reporting/different_lifetimes.rs

```diff
@@ -204,6 +204,13 @@ impl<'a, 'gcx, 'tcx> Visitor<'gcx> for FindNestedTypeVisitor<'a, 'gcx, 'tcx> {
 
     fn visit_ty(&mut self, arg: &'gcx hir::Ty) {
         match arg.node {
+            hir::TyBareFn(_) => {
+                self.depth += 1;
+                intravisit::walk_ty(self, arg);
+                self.depth -= 1;
```

Comment:
No such field `depth`. Do you mean the depth of the de Bruijn index or something?

```rust
[00:06:31] error[E0609]: no field `depth` on type `&mut infer::error_reporting::different_lifetimes::FindNestedTypeVisitor<'a, 'gcx, 'tcx>`
[00:06:31]    --> /checkout/src/librustc/infer/error_reporting/different_lifetimes.rs:208:22
[00:06:31]     |
[00:06:31] 208 |                 self.depth += 1;
[00:06:31]     |                      ^^^^^
[00:06:31] 
[00:06:31] error[E0609]: no field `depth` on type `&mut infer::error_reporting::different_lifetimes::FindNestedTypeVisitor<'a, 'gcx, 'tcx>`
[00:06:31]    --> /checkout/src/librustc/infer/error_reporting/different_lifetimes.rs:210:22
[00:06:31]     |
[00:06:31] 210 |                 self.depth -= 1;
[00:06:31]     |                      ^^^^^
[00:06:31] 
[00:06:41] error: aborting due to 2 previous errors
[00:06:41] 
[00:06:41] error: Could not compile `rustc`.
```